### DISCUSSION TITLE
Addons for DDO

### DISCUSSION THREAD

**Discussion by morrikan** [JuneStepp/OneLauncher#42] [Q&A] [answered]
Is there a developer guide for making Addons? DDO doesn't have LUA support so I'm curious how this works.

**Comment by JuneStepp** [comment, marked as answer]
Currently, the only type of addon supported by DDO are skins. There's not as much info out there as for LOTRO, but the skinning systems should be pretty similar.

I haven't seen a comprehensive guide for DDO, but you can direct questions to the `ui-skins` channel on the [DDO Community Discord](https://discord.gg/ddo).
The basics are to grab the [latest art pack/default skin](https://www.lotrointerface.com/downloads/info1169-DefaultSkinforupdate53.html) and edit any images that you want to be different. You can then include those images in your own custom skin folder along with a `SkinDefinition.xml` file. You should be able to get an idea for how it works if you look at the files for any existing skin, including the default one linked above.

With regard to DDO skins with OneLauncher, it's the same as any other addon. There's more info [in the README](https://github.com/JuneStepp/OneLauncher#add-on-manager-info-for-developers), but unless something fancy is being done, the folder for the custom skin just has to be zipped up and uploaded to [LotroInterface](https://www.lotrointerface.com).

Feel free to reopen this discussion if you have further questions. The DDO discord is probably a better place to ask, though.

**Comment by morrikan** [comment]
I regularly export "default" skin packs from the game client for the skin modders to use as templates. Thanks for following up - I didn't think I had missed something, but I had to check.

**Reply by JuneStepp** [reply]
lol. Should have caught that it was you. OneLauncher does also have some custom Python scripting stuff that'll get expanded in the future, but it only works if the game is opened via OneLauncher, and _probably_ won't get anything all that comparable to the LUA API.

I'm not really involved with the DDO side of things, so if you ever have any ideas or information you think I should include somewhere, feel free to reach out!
